### ISSUE TITLE
Microsoft AD Size ignored

### ISSUE THREAD

**Issue by AndyFelton** [terraform-providers/terraform-provider-aws#2192]
Hi there,

### Terraform Version
Terraform v0.10.7

### Affected Resource(s)
aws_directory_service_directory

### Terraform Configuration Files
```hcl
resource "aws_directory_service_directory" "active-directory" {
  name     = "${var.name}"
  password = "${var.password}"
  type     = "MicrosoftAD"
  size     = "Small"

  vpc_settings {
    vpc_id     = "${var.vpc_id}"
    subnet_ids = ["${var.subnet1}", "${var.subnet2}"]
  }
}
```

### Expected Behavior
I would expect a Microsoft AD to be created on AWS with a Size of Standard

### Actual Behavior
Microsoft AD is created with a Size of Enterprise

### Steps to Reproduce
1. `terraform apply`

**Comment by edwardbartholomew** [comment]
I couldn't find in the AWS docs but there is only one size (Large/Enterprise) for type = MicrosoftAD. If you try to create MicrosoftAD in the console, you will notice there are no options for size.

**Comment by lorengordon** [comment]
I think it's something of a [new feature](https://aws.amazon.com/blogs/security/introducing-aws-directory-service-for-microsoft-active-directory-standard-edition/). You can see the option in the [API docs](http://docs.aws.amazon.com/directoryservice/latest/devguide/API_CreateMicrosoftAD.html#API_CreateMicrosoftAD_RequestSyntax).

**Comment by AndyFelton** [comment]
There is an option for Standard and Enterprise from the console. If you create a standard in the console and import it to terraform it reads as Small. I think it’s also an issue in the Go AWS API.

**Comment by edwardbartholomew** [comment]
I just tried and didn’t have any option called Size like there is for SimpleAD. There is a distinction between Microsoft AD Standard Edition and Microsoft AD Enterprise Edition which I don’t remember from a couple months ago. Thanks for the link lorengordon. Seems size terminology could also refer to edition for type=MicrosoftAD going forward?

**Comment by lorengordon** [comment]
> Seems size terminology could also refer to edition for type=MicrosoftAD going forward?

@edwardbartholomew That's what it seems to me:

* `type=MicrosoftAD` and `size=Small` would be Standard Edition.
* `type=MicrosoftAD` and `size=Large` would be Enterprise Edition.

**Comment by philharle** [comment]
I think this issue is two-fold...
a) We need to pass through `size` as part of the `createActiveDirectoryService` function.
b) Looking through Amazon's API it would seem like the `size` parameter is missing at their end too for MicrosoftAD. Has anyone already raised this as a feature request to AWS?

**Comment by lorengordon** [comment]
@philharle it looks to me like it's present in the API: http://docs.aws.amazon.com/directoryservice/latest/devguide/API_CreateMicrosoftAD.html#DirectoryService-CreateMicrosoftAD-request-Size.

Where are you looking that makes it seem otherwise?

**Comment by philharle** [comment]
@lorengordon Apologies my bad, I had incorrectly assumed Terraform had a dependency on the AWS SDK.
What I should be saying is that AWS haven't added the size parameter into their own Go SDK. It exists for SimpleAD here https://github.com/aws/aws-sdk-go/blob/master/service/directoryservice/api.go#L4694-L4697 but not for MicrosoftAD here https://github.com/aws/aws-sdk-go/blob/master/service/directoryservice/api.go#L4801-L4829

**Comment by lorengordon** [comment]
@philharle you are quite right. I believe the SDK code is auto-generated from the model, https://github.com/aws/aws-sdk-go/blob/master/models/apis/ds/2015-04-16/api-2.json#L901-L915. That will need to be updated with a Size parameter before terraform can address this issue.

**Comment by lorengordon** [comment]
I didn't see an issue on the aws-sdk-go project, so I created one: https://github.com/aws/aws-sdk-go/issues/1664

**Comment by bflad** [comment]
Seems like the SDK just added this functionality in v1.12.58: https://github.com/aws/aws-sdk-go/releases/tag/v1.12.58

**Comment by bflad** [comment]
Ah sorry, edition (standard/enterprise) was added in v1.12.58.

**Comment by bflad** [comment]
Implementing `Edition` just happens to now have an open PR: #3421. 😄

**Comment by bflad** [comment]
This has been merged into master and will be released in v1.10.0 of the AWS provider, likely the end of this week.

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
